src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -258,45 +276,70 @@ class boss_thaddius : public CreatureScript
 
             switch (events.ExecuteEvent())
             {
-                case EVENT_THADDIUS_START:
-                    StartEvent();
+                case EVENT_THADDIUS_INIT:
+                {
+                    me->RemoveAllAuras();
+                    me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
+                    for (SummonList::const_iterator itr = summons.begin(); itr != summons.end(); ++itr)
+                    {
+                        if (Creature* cr = ObjectAccessor::GetCreature(*me, (*itr)))
+                        {
+                            if (cr->GetEntry() == NPC_TESLA_COIL)
+                            {
+                                Unit::Kill(cr, cr);
+                            }
+                        }
+                    }
+                    if (GameObject* go = me->FindNearestGameObject(GO_TESLA_COIL_LEFT, 100.0f))
+                    {
+                        go->SetGoState(GO_STATE_READY);
+                    }
+                    if (GameObject* go = me->FindNearestGameObject(GO_TESLA_COIL_RIGHT, 100.0f))
+                    {
+                        go->SetGoState(GO_STATE_READY);
+                    }
```

Comment:
this kind of if() is repeated a lot of times, you could create an external function and re-use it in all this PR

Reply:
maybe something like creatureAI->SetGoState(GO_TESLA_COIL_RIGHT, GO_TESLA_COIL_RIGHT);

And then inside creatureAI search for the gameobject in 5000.0f

It's not optimal but it's reusable

Reply:
I guess what's best should be creating a Gameobject pointer list/vector that would be initiated on create. Then search for the uint32 id passed onto a function and just activate whatever object necessary.